adafruit_esp32spi/adafruit_esp32spi_wifimanager.py

```diff
@@ -347,3 +353,57 @@ def signal_strength(self):
         if not self.esp.is_connected:
             self.connect()
         return self.esp.ap_info.rssi
+
+
+# pylint: disable=too-many-instance-attributes
+class ESPSPI_WiFiManager(WiFiManager):
+    """
+    A legacy class to help manage the Wifi connection. Please update to using WiFiManager
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        esp,
+        secrets,
+        status_pixel=None,
+        attempts=2,
+        connection_type=WiFiManager.NORMAL,
+        debug=False,
+    ):
+        """
+        :param ESP_SPIcontrol esp: The ESP object we are using
+        :param dict secrets: The WiFi secrets dict
+            The use of secrets.py to populate the secrets dict is depreciated
+            in favor of using settings.toml.
+        :param status_pixel: (Optional) The pixel device - A NeoPixel, DotStar,
+            or RGB LED (default=None). The status LED, if given, turns red when
+            attempting to connect to a Wi-Fi network or create an access point,
+            turning green upon success. Additionally, if given, it will turn blue
+            when attempting an HTTP method or returning IP address, turning off
+            upon success.
+        :type status_pixel: NeoPixel, DotStar, or RGB LED
+        :param int attempts: (Optional) Failed attempts before resetting the ESP32 (default=2)
+        :param const connection_type: (Optional) Type of WiFi connection: NORMAL or ENTERPRISE
+        """
+
+        warnings.warn(
+            "ESP32WiFiManager, which uses `secrets`, is deprecated. Use WifiManager instead and "
+            "fetch values from settings.toml with `os.getenv()`."
+        )
+
+        ssid = secrets.get("ssid")
+        password = secrets.get("secrets", None)
+        enterprise_ident = secrets.get("ent_ident", "")
+        enterprise_user = secrets.get("ent_user")
+        super().__init__(
+            esp=esp,
+            ssid=ssid,
+            password=password,
+            enterprise_ident=enterprise_ident,
+            enterprise_user=enterprise_user,
+            status_pixel=status_pixel,
+            attempts=attempts,
+            connection_type=connection_type,
+            debug=debug,
```

Comment:
Remove intermediate variables to shorten code. I don't think it's any less clear.
```suggestion
        super().__init__(
            esp=esp,
            ssid=secrets.get("ssid"),
            password=secrets.get("secrets"),
            enterprise_ident=secrets.get("ent_ident", ""),
            enterprise_user=secrets.get("ent_user"),
            status_pixel=status_pixel,
            attempts=attempts,
            connection_type=connection_type,
            debug=debug,
```